etna/transforms/timestamp/holiday.py

```diff
@@ -1,20 +1,67 @@
-import datetime
 from enum import Enum
 from typing import List
 from typing import Optional
 
 import holidays
 import pandas as pd
+from pandas.tseries.offsets import MonthBegin
+from pandas.tseries.offsets import MonthEnd
+from pandas.tseries.offsets import QuarterBegin
+from pandas.tseries.offsets import QuarterEnd
+from pandas.tseries.offsets import Week
+from pandas.tseries.offsets import YearBegin
+from pandas.tseries.offsets import YearEnd
+from typing_extensions import assert_never
 
 from etna.datasets import TSDataset
 from etna.transforms.base import IrreversibleTransform
 
+_DEFAULT_FREQ = object()
+
+
+# TODO: it shouldn't be called on freq=None, we should discuss this
```

Comment:
We should discuss how to deal with it. I think it deserves a separate task for fixing.